### ISSUE TITLE
How to use the Hive package in a sync data from server scenario?

### ISSUE THREAD

**Issue by adrian18hd** [hivedb/hive#59]
**Question**
Hi, I have a question related to how to use the hive package. 
Btw, I like it a lot and I'm definitely planning on using it in my projects.
I have a scenario where I need to synchronize some data from server, but I don't want to let the user wait until the sync finishes. 
Also, some stuff can be written/read to/from the box even if the sync did not finished. So I tried to simulate this scenario in a very "stupid way":

**Code sample**
 ```
   Future<void> Write1() async {
    final now = DateTime.now();
    final box = Hive.box("test");
    for (int i = 0; i < 10000; i++) {
      await box.put(i.toString(), i.toString());
    }
    print("Finished 1 after ${DateTime.now().difference(now).inMilliseconds}");
  }

  Future<void> Write2() async {
    final now = DateTime.now();
    final box = Hive.box("test");
    for (int i = 10001; i < 20000; i++) {
      await box.put(i.toString(), i.toString());
    }
    print("Finished 2 after ${DateTime.now().difference(now).inMilliseconds}");
  }

  test() async {
	final box = await Hive.openBox("test");
        unawaited(Write1());
        unawaited(Write2());
        await Future.delayed(Duration(milliseconds: 300));
	print("Getting value from box: ${box.get("100")}");
  }
```

In the first run, the code almost always runs with success. But if I restart the app, in 90% of the cases I get the exception below.
Am I using your library in a wrong way?

```
E/flutter ( 4637): [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: Bad state: No element
E/flutter ( 4637): #0      ListQueue.removeFirst (dart:collection/queue.dart:729:25)
E/flutter ( 4637): #1      Keystore.cancelTransaction (package:hive/src/box/keystore.dart:143:33)
E/flutter ( 4637): #2      BoxImpl._writeFrame (package:hive/src/box/box_impl.dart:74:16)
E/flutter ( 4637): <asynchronous suspension>
E/flutter ( 4637): #3      BoxImpl.put (package:hive/src/box/box_impl.dart:57:12)
E/flutter ( 4637): #4      GlobalApplicationState.Write2 (package:eatntrack/features/shared/presentation/models/global_application_state.dart:56:17)
E/flutter ( 4637): <asynchronous suspension>
E/flutter ( 4637): #5      GlobalApplicationState._initialize (package:eatntrack/features/shared/presentation/models/global_application_state.dart:67:15)
E/flutter ( 4637): #6      _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
E/flutter ( 4637): #7      _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter ( 4637): #8      _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter ( 4637): #9      _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
E/flutter ( 4637): #10     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
E/flutter ( 4637): #11     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
E/flutter ( 4637): #12     Future._completeWithValue (dart:async/future_impl.dart:522:5)
E/flutter ( 4637): #13     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
E/flutter ( 4637): #14     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
E/flutter ( 4637): #15     HiveImpl.openBox (package:hive/src/hive_impl.dart)
E/flutter ( 4637): #16     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
E/flutter ( 4637): #17     _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter ( 4637): #18     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter ( 4637): #19     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
E/flutter ( 4637): #20     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
E/flutter ( 4637): #21     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
E/flutter ( 4637): #22     Future._completeWithValue (dart:async/future_impl.dart:522:5)
E/flutter ( 4637): #23     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
E/flutter ( 4637): #24     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
E/flutter ( 4637): #25     HiveImpl.openBoxInternal (package:hive/src/hive_impl.dart)
E/flutter ( 4637): #26     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
E/flutter ( 4637): #27     _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter ( 4637): #28     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter ( 4637): #29     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
E/flutter ( 4637): #30     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
E/flutter ( 4637): #31     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
E/flutter ( 4637): #32     Future._completeWithValue (dart:async/future_impl.dart:522:5)
E/flutter ( 4637): #33     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
E/flutter ( 4637): #34     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
E/flutter ( 4637): #35     StorageBackendVm.initialize (package:hive/src/backend/storage_backend_vm.dart)
E/flutter ( 4637): #36     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
E/flutter ( 4637): #37     _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter ( 4637): #38     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter ( 4637): #39     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
E/flutter ( 4637): #40     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
E/flutter ( 4637): #41     Future._propagateToListeners (dart:async/future_impl.dart:707:32)
E/flutter ( 4637): #42     Future._completeWithValue (dart:async/future_impl.dart:522:5)
E/flutter ( 4637): #43     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:30:15)
E/flutter ( 4637): #44     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:288:13)
E/flutter ( 4637): #45     Lock.synchronized (package:hive/src/util/lock.dart)
E/flutter ( 4637): #46     _asyncThenWrapperHelper.<anonymous closure> (dart:async-patch/async_patch.dart:71:64)
E/flutter ( 4637): #47     _rootRunUnary (dart:async/zone.dart:1132:38)
E/flutter ( 4637): #48     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
E/flutter ( 4637): #49     _FutureListener.handleValue (dart:async/future_impl.dart:137:18)
E/flutter ( 4637): #50     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:678:45)
```
**Version**
 - Platform: Windows Android Simulator
 - Flutter version: [e.g. 1.9.1]
 - Hive version: [e.g. 1.0.0]

**Comment by leisim** [comment]
This seems to be related to #57 but I could not reproduce it until now. Thanks for your sample, I can reproduce the problem now and I'll fix it asap.

**Comment by adrian18hd** [comment]
That's great, thank you!

**Comment by leisim** [comment]
Sorry this was a bigger problem but it is now fixed with `1.1.0-beta`.
